iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/optimizations/PushPredicateIntoTableScan.java

```diff
@@ -751,6 +756,205 @@ private void appendSortNodeForMergeSortJoin(JoinNode joinNode) {
       joinNode.setRightChild(rightSortNode);
     }
 
+    @Override
+    public PlanNode visitSemiJoin(SemiJoinNode node, RewriteContext context) {
+      Expression inheritedPredicate =
+          context.inheritedPredicate != null ? context.inheritedPredicate : TRUE_LITERAL;
+      if (!extractConjuncts(inheritedPredicate)
+          .contains(node.getSemiJoinOutput().toSymbolReference())) {
+        return visitNonFilteringSemiJoin(node, context);
```

Comment:
Is there always a FilterNode(`expr`) above SemiJoin? So which sql can satisfy this branch?
![image](https://github.com/user-attachments/assets/0b97af37-c085-4dc8-b69b-22e3c6cdc3d1)